### PR TITLE
[ZEPPELIN-1150] Table contents disappear 2nd run onwards

### PR DESCRIPTION
**Pull request by Leemoonsoo**
### What is this PR for?
Fix for [ZEPPELIN-1150](https://issues.apache.org/jira/browse/ZEPPELIN-1150).

### What type of PR is it?
Bug Fix

### Todos
* [x] - Recreate table on data refresh
* [x] - Better solution for [ZEPPELIN-1078](https://issues.apache.org/jira/browse/ZEPPELIN-1078) without performance degrade

### What is the Jira issue?
[ZEPPELIN-1150](https://issues.apache.org/jira/browse/ZEPPELIN-1150)

### How should this be tested?
Reproduce procedure described in the issue

### Questions:
* Does the licenses files need update? no 
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

File: zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js [hunk 1]
```diff
@@ -470,7 +472,7 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
         $scope.renderHtml();
       } else if (newType === 'ANGULAR' && resultRefreshed) {
         $scope.renderAngular();
-      } else if (newType === 'TEXT') {
+      } else if (newType === 'TEXT' && resultRefreshed) {
```
Comment (astroshim) [inline zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js]: Maybe this line causes the https://github.com/apache/zeppelin/pull/1104?

**Comment by astroshim** [comment]
I got a following build error. 
```
[INFO] --- frontend-maven-plugin:0.0.25:grunt (grunt build) @ zeppelin-web ---
[INFO] Running 'grunt build --no-color' in /home/nflabs/tmp/zeppelin/zeppelin-web
[INFO] Running "jscs:all" (jscs) task
[INFO] disallowMultipleLineBreaks: ; and console.log("updateParagraph oldData %o, newData %o. type %o -> %o, mode %o -> %o", $scope.paragraph, data, oldType, newType, oldGraphMode, newGraphMode); should have at most 2 line(s) between them at src/app/notebook/paragraph/paragraph.controller.js :
[INFO]    412 |        isEmpty(data.paragraph.result) !== isEmpty($scope.paragraph.result) ||
[INFO]    413 |        data.paragraph.status === 'ERROR' || (data.paragraph.status === 'FINISHED' && statusChanged) ||
[INFO]    414 |        (!newActiveApp && oldActiveApp !== newActiveApp);
[INFO] -----------------------------------------------------------------^
[INFO]    415 |
[INFO]    416 |
[INFO] >> 1 code style errors found!
```

**Comment by Leemoonsoo** [comment]
@astroshim Thanks for testing. Build error is addressed in last commit.

**Comment by corneadoug** [comment]
@Leemoonsoo Tested,
Fix for the table after re-run confirmed.

For the second item, I would keep the solution used in https://issues.apache.org/jira/browse/ZEPPELIN-1078
Since you changes do not bring any improvement regarding performance degrade.
(We still do a renderText, when the text result is the same as before)

**Comment by astroshim** [comment]
Build error has gone and working good!

**Comment by Leemoonsoo** [comment]
@corneadoug Try revert line 413 and 473. And then run paragraph multiple times (with table result). In my machine after 2~3 times without 413, 473 change, browser almost freeze.

**Comment by Leemoonsoo** [comment]
@corneadoug I tried more and it looks like slowing down without 413 and 473 change is not consistently reproducible.

Anyway, logically, 413 and 473 change reduce text rendering from
4 times (rendering 4 status change FINISHED -> PENDING -> RUNNING -> FINISHED)
to 
1 time (RUNNING -> FINISHED)

So, i'd like to keep it while it doesn't bring any side effect or increase complexity of code.

**Comment by corneadoug** [comment]
@Leemoonsoo I tested again, originally I was testing with previous `|| resultRefreshed`
473 does help the text rendering indeed. However I have the same improvement without 413 for both TABLE and TEXT types

**Comment by Leemoonsoo** [comment]
@corneadoug 413 and 473 "together" are alternative fix for ZEPPELIN-1078 for better performance.

Without 413, applying 473 will bring ZEPPELIN-1078 problem again.

**Comment by corneadoug** [comment]
@Leemoonsoo I was testing with %spark only, while ZEPPELIN-1078's issue shows with %python.
LGTM

**Comment by Leemoonsoo** [comment]
Thanks @astroshim @corneadoug for review this!

Merge it into master if there're no more discussions.

**Comment by prabhjyotsingh** [comment]
I missed this (which solve the same problem as https://github.com/apache/zeppelin/pull/1182), let me check this.

**Comment by prabhjyotsingh** [comment]
👍

**Pull request closed without merging**